pkg/daemon/ceph/client/erasure-code-profile.go

```diff
@@ -89,7 +89,7 @@ func CreateErasureCodeProfile(context *clusterd.Context, clusterInfo *ClusterInf
 		profilePairs = append(profilePairs, fmt.Sprintf("crush-device-class=%s", pool.DeviceClass))
 	}
 
-	args := []string{"osd", "erasure-code-profile", "set", profileName}
+	args := []string{"osd", "erasure-code-profile", "set", profileName, "--force"}
```

Comment:
Instead of force creating the profile, can we delete the EC profile during removal of the pool? Then it would be more clean for the uninstall case.

Reply:
I don't see `--force` arg in the removal command. `--force` is passed while creating only [here](https://docs.ceph.com/en/latest/rados/operations/erasure-code-profile/#osd-erasure-code-profile-set).

Reply:
Instead of passing the `--force` flag during creation, I am asking about a different approach. The root issue is that the EC profile is not deleted when the pool is deleted. If we deleted the EC profile during pool deletion, then the uninstall would be properly cleaned up and there would be no need for the `--force` flag during pool creation.

Reply:
I see the profile is deleted while pool creation here, https://github.com/rook/rook/blob/40e66332801d8b7165427ec0dd7caec1d4331628/pkg/operator/ceph/object/objectstore.go#L604-L612

Reply:
Ok great. So does the original issue #2016 repro anymore? Maybe it's already fixed. If force is needed during new pool creation, seems like something wasn't cleaned up during pool removal.

Reply:
I did test my changes and it was working...I tried both with YAML and cli so I believe it's working.
My first thought was the same to see if we are removing the profile :)

Reply:
And without your changes it was failing? I'd like to understand what wasn't cleaned up during pool deletion.

Reply:
Yes, it was failing with the same error mentioned in the issue...not sure what wasn't cleared because both pool and profile are getting deleted but we have to use `--force` when recreating with a different setting and in the Ceph docs also it was mentioned to use `--force`.

Reply:
Ok thanks for the confirmation that you see the pool and profile both cleaned up. Sounds good to go with `--force`. It also works for new pool creation, right? We just don't have an integration test for EC pools so wanted to double check.

Reply:
okay...give me some time I'll paste log/output here.

Reply:
created `pool-ec.yaml` 
``` erasureCoded:
    dataChunks: 2
    codingChunks: 1
```
```
2021-05-05 14:28:01.268197 I | cephclient: setting pool property "allow_ec_overwrites" to "true" on pool "ec-pool"
2021-05-05 14:28:02.294373 I | cephclient: setting pool property "compression_mode" to "none" on pool "ec-pool"
2021-05-05 14:28:04.287084 I | cephclient: creating EC pool ec-pool succeeded
2021-05-05 14:28:13.784876 I | op-mon: parsing mon endpoints: a=10.101.111.169:6789
```
deleting `pool-ec.yaml`
```
2021-05-05 14:28:13.784899 I | ceph-block-pool-controller: deleting pool "ec-pool"
2021-05-05 14:28:14.321419 I | cephclient: no images/snapshosts present in pool "ec-pool"
2021-05-05 14:28:14.321474 I | cephclient: purging pool "ec-pool" (id=2)
2021-05-05 14:28:16.312577 I | cephclient: purge completed for pool "ec-pool"
2021-05-05 14:28:16.872855 I | ceph-spec: removing finalizer "cephblockpool.ceph.rook.io" on "ec-pool"

creating `pool-ec.yaml` with
```
  erasureCoded:
    dataChunks: 4
    codingChunks: 2
```
```
2021-05-05 14:28:36.079002 I | ceph-spec: adding finalizer "cephblockpool.ceph.rook.io" on "ec-pool"
2021-05-05 14:28:36.086710 W | ceph-block-pool-controller: failed to set pool "ec-pool" status to "Progressing". failed to update object "ec-pool" status: Operation cannot be fulfilled on cephblockpools.ceph.rook.io "ec-pool": the object has been modified; please apply your changes to the latest version and try again
2021-05-05 14:28:36.092343 I | op-mon: parsing mon endpoints: a=10.101.111.169:6789
2021-05-05 14:28:36.383661 I | ceph-block-pool-controller: creating pool "ec-pool" in namespace "rook-ceph"
2021-05-05 14:28:39.364564 I | cephclient: setting pool property "allow_ec_overwrites" to "true" on pool "ec-pool"
2021-05-05 14:28:40.429747 I | cephclient: setting pool property "compression_mode" to "none" on pool "ec-pool"
2021-05-05 14:28:42.403420 I | cephclient: creating EC pool ec-pool succeeded
2021-05-05 14:28:43.238145 I | clusterdisruption-controller: all osds are up. last drained failure domain: "".  pg health: "cluster is not fully clean. PGs: [{StateName:unknown Count:32} {StateName:active+clean Count:1}]". waiting for PGs to be active+clean or PG health check to timeout
2021-05-05 14:28:52.037688 I | op-mon: parsing mon endpoints: a=10.101.111.169:6789
```
deleting `pool-ec.yaml`
```
2021-05-05 14:28:52.037774 I | ceph-block-pool-controller: deleting pool "ec-pool"
2021-05-05 14:28:52.368300 I | clusterdisruption-controller: all osds are up. last drained failure domain: "".  pg health: "cluster is not fully clean. PGs: [{StateName:active+undersized Count:32} {StateName:active+clean Count:1}]". waiting for PGs to be active+clean or PG health check to timeout
2021-05-05 14:28:52.621629 I | cephclient: no images/snapshosts present in pool "ec-pool"
2021-05-05 14:28:52.622781 I | cephclient: purging pool "ec-pool" (id=3)
2021-05-05 14:28:54.434969 I | cephclient: purge completed for pool "ec-pool"
2021-05-05 14:28:55.011856 I | ceph-spec: removing finalizer "cephblockpool.ceph.rook.io" on "ec-pool"
```